src/lib/Microsoft.Health.Extensions.Fhir.R4/Telemetry/Exceptions/FhirServiceExceptionProcessor.cs

```diff
@@ -74,17 +75,6 @@ public static (Exception customException, string errorName) CustomizeException(E
                     errorName = nameof(FhirServiceErrorCode.ConfigurationError);
                     return (new InvalidFhirServiceException(message, exception, errorName), errorName);
 
-                case HttpRequestException _:
-                    // TODO: In .NET 5 and later, check HttpRequestException's StatusCode property instead of the Message property
-                    if (exception.Message.Contains(FhirResources.HttpRequestErrorNotKnown, StringComparison.CurrentCultureIgnoreCase))
```

Comment:
In my testing with the previous FhirClient, the `HttpRequestException` given when the FHIR service in the FHIR service URL does not exist, e.g. _https://[non-existent FHIR service name].azurehealthcareapis.com_, has a null `StatusCode`. (This is why I still checked the `Message` in the closed PR #168.)

So if this scenario with a non-existent FHIR service still produces a null `StatusCode` with the new FhirClient, then
- checking the `Message` shouldn't be handled as part of the `NotFound` `StatusCode` (via `DetermineNotFoundErrorMessage`), and
- the `NotFound` case should continue producing the exception `message` of `FhirResources.FhirServiceNotFound`.

Reply:
Thanks for reviewing Carrie! 

Yesterday, I ran through the test cases in the document you sent over and found that the FhirClient is throwing a FhirException which also inherits from the base class System.Exception - similar to the FhirOperationException we recognized previously. We can leverage FhirException in place of FhirOperationException and maintain all of the current logic in the FhirServiceExceptionProcessor. I also incorporated the changes from your PR for the HttpRequestException cases and updated the FhirServiceExceptionProcessor tests with your changes.

Please look on the latest changes and provide your feedback

Reply:
The changes to FhirServiceExceptionProcessor and corresponding unit tests look good. Thanks for testing!